### DISCUSSION TITLE
LinkNeighborLoader: How can I sample source2target edges connected seed node as source node?

### DISCUSSION THREAD

**Discussion by haru-256** [pyg-team/pytorch_geometric#9816] [Q&A]
I want to sample source2target edges connected to seed node as source by `LinkNeighborLoader`.
For example, in the below image, I want to sample edges: [0, 4] and [0, 5] by `LinkNeighborLoader`. But `LinkNeighborLoader` samples only edges to seed node, such as [1, 0], [2, 0], [3, 0].

![src2tgt](https://github.com/user-attachments/assets/b0dd7f8c-e23f-42c9-8b64-d6bd0450d007)

How can I achive this?

**Comment by Romain-Nicolle** [comment]
Hi,

I am not an expert so it might be a stupid idea but couldn't you create the reverse edge_type and sample from it ?
For example, I assume you have something like that in your data object:

 ```
data['node_A', 'edge_type_A', 'node_A'].edge_index = ....
```

You could create the opposite edge_type by doing:

```
data['node_A', 'rev_edge_type_A', 'node_A'].edge_index = data['node_A', 'edge_type_A', 'node_A'].edge_index.flip(dims = (0,))
```

That way, in this new edge_type, the src and dst nodes are reversed comparatively to the first edge_type.
This might require that you use a HeteroData object and not a Data object though.
Maybe you can create a temporary tensor outside of the data object ? Or reverse the src and dst somewhere during your script to make it work.

I am sure there are smarter ways to do it, but I think/hope this sould work.

Good luck.
